### ISSUE TITLE
Geocoding: Add Census tract to the building_permit table during creation

### ISSUE THREAD

**Issue by NealHumphrey** [codefordc/housing-insights#177]
The building_permit records have been geocoded by the city, but they do not include the census tract of the location. 

Use the `MARADDRESSREPOSITORYID` field of the incoming data and the `MarApiConn` class to add this geocoding to the building_permit table before it is written to S3. 

We want to be able to handle new MAR addresses that have not previously been geocoded, but we also want to cache our geocoding so that we do not need to query the API every time a new developer rebuilds their local copy of the database. The proposed approach below tries to meet both those goals, but feel free to suggest different approaches if you have them. 

- [ ] Read all records of the database building_permits table and output a mapping of mar_id to census_tract. Output this as a flat csv file to a folder in `/data/raw/geographic_data/mar_mapping` (note, the census_tract_mapping that already exists in geograph_data is for converting different formats of census_tract name - the unique ID in this new table will be MAR so mar_mapping is a better place).

    Be sure to build in a sleep function, and the default behavior should be to only encode the first 10 or so records, with a user option to encode all - this is so we don't overload the MAR server and stay on good terms with OCTO. Useful existing code:
    - `/python/housinginsights/tools/dbtools.py` has methods for getting connections to the database
    -  [Read about SQLAlchemy session,connection, and enginge](http://stackoverflow.com/questions/34322471/sqlalchemy-engine-connection-and-session-difference/42772654#42772654) if you're not familiar with them. You'll probably want to use a connection. 
    - `/python/housinginsights/tools/ingestion/SQLWriter.py` has a `write_file_to_sql` method that demonstrates use of a connection in a transaction. Not the same thing as what we're trying to do but may be useful for reference if you're not familiar with SQLAlchemy. 

**Pull request after step 1 is a good idea!**

- [ ] Add a geocode method to the base Cleaner class in `/ingestion/Cleaners.py`. 
    - Check for the existence of the appropriate mapping file in `mar_mapping`; if it doesn't exist, try downloading it from S3.
    - Check if the mar exists in our flat file. If so, add it to the row dictionary as `row['census_tract']`. If not, use a MarApiConn to get it and add it to both the row and our flat file. Add a (very short) sleep command to getting any unknown records. 

This step will depend on issue #176 to be complete for it to work properly - right now we do not have a way for fields that do not exist in the source CSV file to be added to the table.

**Comment by NealHumphrey** [comment]
Putting this on hold - we may be switching to the opendata.dc.gov version of the building permits data, which looks like it already includes census tract. We likely still need this methodology for a different dataset, though.
